calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -0,0 +1,324 @@
+from calico_ctl.pool import *
+from calico_ctl.pool import validate_arguments as pool_validate_arguments
+from calico_ctl.profile import validate_arguments as profile_validate_arguments
+from calico_ctl.container import validate_arguments as container_validate_arguments
+from pycalico.datastore_datatypes import BGPPeer
+
+
+class TestBgp(unittest.TestCase):
+
+    @patch('calico_ctl.bgp.sys.exit',autospec=True)
+    def test_validate_arguments(self, m_sys_exit):
+        """
+        Test validate_arguments for calicoctl bgp command
+        """
+        # Set up arguments
+        cases = (({'<PEER_IP>':'127.a.0.1'},False),
+                 ({'<PEER_IP>':'aa:bb::zz'},False),
+                 ({'<AS_NUM>':9},True),
+                 ({'<AS_NUM>':'9'},True),
+                 ({'<AS_NUM>':'nine'},False),
+                 ({'show':1,'--ipv4':1},True))
+
+        # Call method under test for each test case
+        # Assert that method exits on bad input
+        for case, is_valid in cases:
+            print "Testing case %s ..." % case
+            bgp_validate_arguments(case)
+            if is_valid:
+                assert not m_sys_exit.called
+            else:
+                assert m_sys_exit.called
+            m_sys_exit.reset_mock()
+
+    @patch('calico_ctl.bgp.check_ip_version',autospec=True)
+    @patch('calico_ctl.bgp.BGPPeer',autospec=True)
+    @patch('calico_ctl.bgp.client',autospec=True)
+    def test_bgp_peer_add(self,m_client,m_BGPPeer,m_check_ipv):
+        """
+        Test bgp_peer_add function for calico_ctl bgp
+        """
+        # Set up mock objects
+        address = '1.2.3.4'
+        peer =  Mock(spec=BGPPeer)
+        m_check_ipv.return_value = address
+        m_BGPPeer.return_value = peer
+
+        # Call method under test
+        bgp_peer_add('1.2.3.4','v4',1)
+
+        # Assert
+        m_check_ipv.assert_called_once_with('1.2.3.4','v4',IPAddress)
+        m_BGPPeer.assert_called_once_with(address,1)
+        m_client.add_bgp_peer.assert_called_once_with('v4',peer)
+
+    @patch('calico_ctl.bgp.check_ip_version',autospec=True)
+    @patch('calico_ctl.bgp.client',autospec=True)
+    def test_bgp_peer_remove(self,m_client,m_check_ipv):
+        """
+        Test bgp_peer_remove function for calicoctl bgp
+        """
+        # Set up mock objects
+        address = '1.2.3.4'
+        m_check_ipv.return_value = address
+
+        # Call method under test
+        bgp_peer_remove('1.2.3.0','v4')
+
+        # Assert
+        m_check_ipv.assert_called_once_with('1.2.3.0','v4',IPAddress)
+        m_client.remove_bgp_peer.assert_called_once_with('v4',address)
+
+    @patch('calico_ctl.bgp.client',autospec=True)
+    def test_set_default_node_as(self,m_client):
+        """
+        Test set_default_node_as function for calicoctl bgp
+        """
+        # Call method under test
+        set_default_node_as(1)
+
+        # Assert
+        m_client.set_default_node_as.\
+            assert_called_once_with(1)
```

Comment:
This doesn't look like it goes over 80 lines, so I would put this on one line.

If it does go over 80, I would write it a bit differently to avoid the \
You can use a variable pointing to the function being called then assert that the variable is called:
```
client_set_def = m_client.set_default_node_as
client_set_def.assert_called_once_with(1)
```